src/awx_plugins/credentials/github_app.py

```diff
@@ -0,0 +1,208 @@
+        },
+    ],
+    'required': ['app_id', 'install_id', 'ssh_key_data'],
+}
+
+github_app_token = ManagedCredentialType(  # type: ignore[misc]
+    namespace='github_app_token',
+    # scm source control management so it can be used for github
+    # authentication.
+    kind='scm',
+    name=_('GitHub App Token'),  # type: ignore[misc]
+    managed=True,
+    inputs=github_app_inputs,
+    custom_injectors=github_app_injector,
+)
+
+
+class GitHubAppBackendArgs(TypedDict, total=False):
+    github_url: str
+    app_id: str | int
+    install_id: str | int
+    ssh_key_data: str
+    jwt_expiry: str | int
+
+
+def github_app_backend(**kwargs: GitHubAppBackendArgs) -> str:
+    """Generate an authentication token for a GitHub App using provided
+    credentials.
+
+    Args:
+        github_url (str): The GitHub instance URL.
+        app_id (str): The GitHub App ID.
+        install_id (str): The installation ID for the GitHub App.
+        ssh_key_data (str): The private key associated with the GitHub App.
+        jwt_expiry (int, optional): JWT expiration time in seconds (default: 600).
+
+    Returns:
+        str: A GitHub App authentication token.
+
+    Raises:
+        ValueError: If any required parameters are missing or invalid.
+    """
+
+    # because the calling functions are not linted, avoid type checking
+    github_url: str | None = kwargs.get(
+        'github_url')  # type: ignore[assignment]
+    app_id: str | None = kwargs.get('app_id')  # type: ignore[assignment]
+    install_id: str | None = kwargs.get(
+        'install_id')  # type: ignore[assignment]
+    ssh_key_data: str | None = kwargs.get(
+        'ssh_key_data')  # type: ignore[assignment]
+    jwt_expiry: str | None = kwargs.get(
+        'jwt_expiry')  # type: ignore[assignment]
+
+    missing_parameters: list[str] = []
+    if not github_url:
+        missing_parameters.append('GitHub URL')
+    if not install_id:
+        missing_parameters.append('Installation ID')
+    if not app_id:
+        missing_parameters.append('Application ID')
+    if not ssh_key_data:
+        missing_parameters.append('Private Key')
+    if missing_parameters:
+        raise ValueError(f'Missing Parameter: {missing_parameters}')
+
+    # Verify that app_id and installation_id are integers
+    try:
+        # already checked for None but linter doesn't know
+        app_id_int: int = int(app_id)  # type: ignore[arg-type]
+        install_id_int: int = int(install_id)  # type: ignore[arg-type]
+    except ValueError as val_err:
+        raise ValueError(
+            f'App ID and Installation ID must be integers {val_err}',
+        ) from val_err
+
+    try:
+        jwt_expiry_int: int = int(jwt_expiry or 600)
+    except ValueError as val_err:
+        raise ValueError(
+            f'JWT Expiry must be an integer {jwt_expiry}',
+        ) from val_err
+
+    auth = Auth.AppAuth(
+        app_id_int, str(ssh_key_data), jwt_expiry=jwt_expiry_int,
```

Comment:
it should already be a string by this point
```suggestion
        app_id_int, private_rsa_key, jwt_expiry=jwt_expiry_int,
```